crates/kubelet/src/volume/mod.rs

```diff
@@ -243,3 +158,78 @@ impl From<Option<String>> for ItemMount {
         }
     }
 }
+
+/// This is a gnarly function to check all of the supported data members of the
+/// Volume struct. Because it isn't a HashMap, we need to check all fields
+/// individually
+async fn configure(
+    vol: &KubeVolume,
+    namespace: &str,
+    client: &kube::Client,
+    plugin_registry: Arc<PluginRegistry>,
+    path: &PathBuf,
+) -> anyhow::Result<Type> {
+    if let Some(cm) = &vol.config_map {
+        let name = &cm
+            .name
+            .as_ref()
+            .ok_or_else(|| anyhow::anyhow!("no configmap name was given"))?;
+        let cm_client: Api<ConfigMap> = Api::namespaced(client.clone(), namespace);
```

Comment:
Should fetching things from the API be moved into the respective modules?

Reply:
I'd prefer that the functions deal with the concrete type rather than all the kube client fetching and whatnot. That way the transition from this design to #482 will be MUCH easier.

That being said let's go forward with this design for now and see if we need to revisit it after we start tackling #482. Either way this is all private functions so it doesn't really affect the public API at this point :)